src/modules/esl-utils/async/debounce.ts

```diff
@@ -37,7 +41,9 @@ export function debounce<F extends AnyToAnyFnSignature>(fn: F, wait = 10, thisAr
       deferred && deferred.resolve(result);
       deferred = null;
     }, wait);
+    return deferred.promise;
```

Comment:
Why do we change the actual return type?
It was made by request to make lazy creation of promise as the user does not need it in 95% of cases.